src/control/cmd/ddb/commands_wrapper.go

```diff
@@ -17,242 +18,269 @@ import (
 */
 import "C"
 
-func ErrorString(errno int) string {
+// DdbContext structure for wrapping the C code context structure
+type DdbContext struct {
+	ctx C.struct_ddb_ctx
+}
+
+// Init initialize DAOS and the context for doing ddb operations
+func (ctx *DdbContext) Init() error {
+	// Must lock to OS thread because vos init/fini uses ABT init and finalize which must be called on the same thread
+	runtime.LockOSThread()
+
+	result := C.ddb_init()
+
+	if result != 0 {
+		return fmt.Errorf(errorString(int(result)))
+	}
+	C.ddb_ctx_init(&ctx.ctx) // Initialize with ctx default values
+
+	return nil
+}
+
+// Fini clean up from Init
+func (ctx *DdbContext) Fini() {
+	C.ddb_fini()
+	runtime.UnlockOSThread()
+}
+
+func errorString(errno int) string {
```

Comment:
Commenting on your other PR reminded me that we have some logic for converting a DER error code to a Go `error` type. You can do this by casting the int to a `daos.Status` type. `daos.Status` conforms to the `error` interface so it can be used directly instead of having to use `fmt.Errorf` on the string.

Reply:
Thanks!! Updated.